hera_cal/lstbin.py

```diff
@@ -0,0 +1,883 @@
+        std_file = "zen.{}".format(file_ext.format('.'.join(pols), "STD", bin_lst[0]))
+        num_file = "zen.{}".format(file_ext.format('.'.join(pols), "NUM", bin_lst[0]))
+
+        # check for overwrite
+        if os.path.exists(bin_file) and overwrite is False:
+            abscal.echo("{} exists, not overwriting".format(bin_file), verbose=verbose)
+            continue
+
+        # write to file
+        utils.data_to_miriad(bin_file, bin_data, bin_lst, freq_array, antpos, flags=flag_data, verbose=verbose, **miriad_kwargs)
+        utils.data_to_miriad(std_file, std_data, bin_lst, freq_array, antpos, verbose=verbose, **miriad_kwargs)
+        utils.data_to_miriad(num_file, num_data, bin_lst, freq_array, antpos, verbose=verbose, **miriad_kwargs)
+
+        del bin_file, std_file, num_file, bin_data, std_data, num_data, bin_lst, flag_data
+        garbage_collector.collect()
+
+
+def make_lst_grid(dlst, lst_start=None, verbose=True):
+    """
+    Make a uniform grid in local sidereal time spanning 2pi radians.
+
+    Parameters:
+    -----------
+    dlst : type=float, delta-LST: width of a single LST bin in radians. 2pi must be equally divisible 
+                by dlst. If not, will default to the closest dlst that satisfies this criterion that
+                is also greater than the input dlst. There is a minimum allowed dlst of 6.283e-6 radians,
+                or .0864 seconds.
+
+    lst_start : type=float, starting point for lst_grid, extending out 2pi from lst_start.
+                            lst_start must fall exactly on an LST bin given a dlst. If not, it is
+                            replaced with the closest bin. Default is lst_start at zero radians.
+
+    Output:
+    -------
+    lst_grid : type=ndarray, dtype=float, uniform LST grid marking the center of each LST bin
+    """
+    # check 2pi is equally divisible by dlst
+    if (np.isclose((2*np.pi / dlst) % 1, 0.0, atol=1e-5) is False) and (np.isclose((2*np.pi / dlst) % 1, 1.0, atol=1e-5) is False):
+        # generate array of appropriate dlsts
+        dlsts = 2*np.pi / np.arange(1, 1000000).astype(np.float)
+
+        # get dlsts closest to dlst, but also greater than dlst
+        dlst_diff = dlsts - dlst
+        dlst_diff[dlst_diff < 0] = 10
+        new_dlst = dlsts[np.argmin(dlst_diff)]
+        abscal.echo("2pi is not equally divisible by input dlst ({:.16f}) at 1 part in 1e5.\n"
+                    "Using {:.16f} instead.".format(dlst, new_dlst), verbose=verbose)
+        dlst = new_dlst
+
+    # make an lst grid from [0, 2pi), with the first bin having a left-edge at 0 radians.
+    lst_grid = np.arange(0, 2*np.pi-1e-7, dlst) + dlst / 2
+
+    # shift grid by lst_start
+    if lst_start is not None:
+        lst_start = lst_grid[np.argmin(np.abs(lst_grid - lst_start))] - dlst/2
+        lst_grid += lst_start
+
+    return lst_grid
+
+
+def lst_rephase(data, bls, freqs, dlst, lat=-30.72152):
```

Comment:
If you want to put this in the PR, but you're not 100% confident of it's functionality, at least make an issue to check that it's working properly.